setup.py

```diff
@@ -4,53 +4,83 @@
 from __future__ import print_function
 
 from distutils.core import setup, Extension
+import sys
+
 try:
     import commands
 except ImportError:
     import subprocess as commands
-import sys
+run = commands.getstatusoutput
 
 version = '0.12'
 
-def pkgconfig(pkg):
-    def _str2list(pkgstr, onlystr):
+
+class PkgConfigExtension(Extension):
+    '''Extension with lazy properties taken from pkg-config'''
+    def __init__(self, *args, **kwargs):
+        self._pkg = kwargs['pkg']
+        del kwargs['pkg']
+        if 'extra_libraries' in kwargs:
```

Comment:
Ok, I see that you are using `kwargs` on line 29 so deleting `extra_libraries` is probably necessary and you can ignore my previous comment.